### DISCUSSION TITLE
Relationship between cryptsetup and ZBM? (How to eliminate cryptsetup error messages during initramfs updates?)

### DISCUSSION THREAD

**Discussion by Binarus** [zbm-dev/zfsbootmenu#666] [Questions] [answered]
Dear all,

although I'm using it since quite a time on multiple machines, I am still excited about ZBM and about how super responsive the developers and the community are. At my very first try, I had found a small glitch that had been fixed in no time, and I never had a real problem since.

However, there is one little thing that I'd like to solve now. Every time my systems (all Debian bookworm) generate the initramfs, I get messages like the following in the output:
```
cryptsetup: ERROR: Couldn't resolve device rpool/sys-gaia
cryptsetup: WARNING: Couldn't determine root device
```

I am using root on ZFS, if that matters.

While I know that I can safely ignore these messages, I'd like to eliminate them. Apparently, the root cause is the package `cryptsetup-initramfs` which eventually got installed when preparing my systems for ZBM the first time. To install ZBM, ~the manual recommends~ the tutorial I was following at that time included the installation of the `cryptsetup` package (which probably caused `cryptsetup-initramfs` to be installed as well). Unfortunately, I can't find that tutorial any more.

The obvious solution to eliminate the above error messages would be to uninstall `cryptsetup-initramfs`. However, this is not my field, so I am afraid that I'll break my systems if I do that.

Therefore, I'd like to learn ~why~ whether ZBM really needs `cryptsetup` at all, and if I can indeed safely remove `cryptsetup-initramfs`.

Thanks a lot in advance!

**Comment by zdykstra** [comment, marked as answer]
ZFSBootMenu doesn't actually support cryptsetup (or LUKS) at all right now. There shouldn't be any complications removing the `cryptsetup-initramfs` package from your system.

**Reply by Binarus** [reply]
Thank you very much!

In the meantime, motivated by your comment, I have removed the `cryptsetup-initramfs` package. This indeed made the error messages go away and did not cause side effects.

I guess I'll now close this "issue".

Thanks again for ZBM! It is a real life saver.

**Comment by ahesford** [comment]
What manual recommends adding the cryptsetup module to the initramfs image?

**Comment by Binarus** [comment]
@ahesford

No manual directly recommends adding the cryptsetup module to the initramfs. However, when I tried to install ZBM for the very first time a while ago, there was a recommendation to install the `cryptsetup` package as a prerequisite for ZBM. Actually, I can't find this reference any longer, but there are a few scripts that are meant to simplify ZBM installation and that still install `cryptsetup`; examples are

https://github.com/Sithuk/ubuntu-server-zfsbootmenu/blob/main/ubuntu_server_encrypted_root_zfs.sh
https://gitlab.com/-/snippets/2099921

I didn't look thoroughly into these sources and thus don't know why exactly these scripts install `cryptsetup`. But it is clear that the scripts a) aim at installing ZBM and that they b) pull in the `cryptsetup` package. It seems that there is *some* sort of slight relationship between `cryptsetup` and ZBM.

Coming back to the subject, as mentioned above, when installing ZBM the very first time, I followed a procedure (probably in some wiki) that contained the following line which also caused `cryptsetup` to be installed:

apt-get install -y ... cryptsetup ...

I *believe* that this in my case could have pulled in `cryptsetup-initramfs` as well.

But the "problem" is solved in the meantime. I have removed the `cryptsetup-initramfs` package. This made the error messages go away and did not cause any side effects AFAICT.

Thank you very much for asking. I have updated my first post accordingly.

**Comment by Binarus** [comment]
The issue has been resolved. As it seems, ZBM neither needs `cryptsetup` nor `cryptsetup-initramfs`, and since having removed the latter, no more error messages have appeared when generating the initramfs. I did not observe negative side effects.
